codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/HttpAuthDecorator.kt

```diff
@@ -86,6 +90,50 @@ class HttpAuthDecorator : ClientCodegenDecorator {
     override val name: String get() = "HttpAuthDecorator"
     override val order: Byte = 0
 
+    override fun authSchemeOptions(
+        codegenContext: ClientCodegenContext,
+        baseAuthSchemeOptions: List<AuthSchemeOptionV2>,
+    ): List<AuthSchemeOptionV2> {
+        val serviceIndex = ServiceIndex.of(codegenContext.model)
+        val authSchemes = serviceIndex.getAuthSchemes(codegenContext.serviceShape)
+        val options = ArrayList<AuthSchemeOptionV2>()
+        for (authScheme in authSchemes.keys) {
+            fun addOption(
+                schemeShapeId: ShapeId,
+                name: String,
+            ) {
+                options.add(
+                    object : AuthSchemeOptionV2 {
+                        override val authSchemeId = schemeShapeId
+
+                        override fun render(
+                            codegenContext: ClientCodegenContext,
+                            operation: OperationShape?,
+                        ) = writable {
+                            rustTemplate(
+                                """
+                                #{AuthSchemeOption}::builder()
+                                    .scheme_id($name)
+                                    .build()
+                                    .expect("required fields set")
```

Comment:
Considering if we should implement `From<AuthSchemeId> for AuthSchemeOption` for better ergonomics.